### DISCUSSION TITLE
Boot files order dependency and Loading plugin

### DISCUSSION THREAD

**Discussion by lucianobosco** [quasarframework/quasar#14359] [General - Components / Directives / etc] [answered]
I have a few Boot files for setting up some stuff:
```javascript
boot: [
      'i18n',
      'notify-defaults',
      'custom-components',
      'http',
      'axios',
      'websockets'
],
```
The `http.js` boot file includes the [websanova vue auth](https://github.com/websanova/vue-auth) package that handles everything related to the JWT authentication. 
It hits an API endpoint when mounted and keeps the tokens/user data within it's `auth` instance.
```javascript
import { boot } from 'quasar/wrappers'
...
import {createAuth} from '@websanova/vue-auth'
import axios from 'axios'

var auth = null

export default boot(({ router }) => {
  auth = createAuth({
    plugins: {
      http: axios,
      router: router
    },
    ...
  })
})

export { auth }

```

The `axios.js` is heavily coupled to `http.js` boot file. It depends on the `auth` instance for setting up the `Authorization` header for all axios request.
It waits for `auth.load()` promise which means that tokens are ready, then uses this token to set `api` instance Authorization header.
```javascript
import { boot } from 'quasar/wrappers'
import { auth } from 'boot/http'
import axios from 'axios'

const baseUrl = import.meta.env.VITE_API_URL

const api = axios.create({
  baseURL: baseUrl,
  headers: {
    common: {
      'X-Requested-With': 'XMLHttpRequest',
      'Access-Control-Allow-Origin': '*'
    }
  }
})

export default boot(({ app, router, store }) => {
  // Wait until tokens/user are ready
  auth.load()
    .then(() => {
      // Set Bearer token for all axios requests
      api.defaults.headers.common.Authorization = `Bearer ${auth.token()}`
    })
})

export { api }
```
In the same way, the `websockets.js` boot file has to wait for `auth.load()` in order to subscribe Pusher private channels with Bearer token and user id provided by `auth.user().id`.

So my questions are:
1- How can I be sure that boot files will be booting always in the same order? How should I handle dependencies between boot files?
I'm seeing that in fact it's always in the same order, but not sure why Quasar is using this order, and if can rely on it.
This is my console log:
```
[vite] connecting...
[vite] connected.
Booting http...
[Quasar] Running SPA.
Booting notify-defaults...
Booting axios...
Booting i18n...
Booting websockets...
Booting custom-components...
``` 

2- I want to use the [Loading plugin](https://quasar.dev/quasar-plugins/loading#introduction) . Where should I place the `.show()` method? Is correct to create a new Boot file for it and trigger the `.hide()` method within my `App.vue` component? I'd like to show the spinners as soon as possible and hide it once my App is mounted.

Any advice will be appreciated

**Comment by metalsadman** [comment, marked as answer]
1. Async boot file https://quasar.dev/quasar-cli-vite/boot-files#anatomy-of-a-boot-file
2. You can use loading in the boot file it self.
